### PR TITLE
refactor: remove now redundant `mark as read on click` functionality

### LINKED ISSUE #712: Remove "Mark as read on click" Setting, which has no effect now
See https://github.com/gitify-app/gitify/pull/706#issuecomment-1900315876

### PR DESCRIPTION
**Pull request by setchy**
Closes #712

### CODE REVIEW

File: src/components/NotificationRow.tsx [hunk 1]
```diff
@@ -84,20 +84,6 @@ export const NotificationRow: React.FC<IProps> = ({
...
-          title="Mark as Read"
-          onClick={() => markNotification(notification.id, hostname)}
-        >
-          <CheckIcon
-            className="hover:text-green-500"
-            size={20}
-            aria-label="Mark as Read"
-          />
-        </button>
-      </div>
```
Comment (adufr) [inline src/components/NotificationRow.tsx:100]: Why remove the `Mark as Read` button?

I agree that the setting to `Mark as Read on click` is now useless, but IMO we should keep the button as a way of clearing a notification without having to actually open it in Github!

Comment (setchy) [reply src/components/NotificationRow.tsx:100]: Good catch - will correct @adufr

**Review by bmulholland — APPROVED** [review]

**Comment by bmulholland** [comment]
That was fast! Thanks :)

**Review by afonsojramos — COMMENTED** [review]
Still catching up on everything! Alright, looking good!

**Pull request merged**